app/src/main/java/com/example/lesson05androidhwcalculator/SCalculatorFragment.kt

```diff
@@ -70,64 +76,156 @@ class SCalculatorFragment : Fragment() {
-                displayField.text = string
+            buttonDiv.setOnClickListener {
+                if (string.isEmpty()) {
+                    init()
+                } else if (string.takeLast(1) !in "./*+-") {
+                    string += "/"
+                    displayField.text = string
+                }
             }
-            button6.setOnClickListener {
-                string += "6"
-                displayField.text = string
+            buttonMult.setOnClickListener {
+                if (string.isEmpty()) {
+                    init()
+                } else if (string.takeLast(1) !in "./*+-") {
+                    string += "*"
+                    displayField.text = string
+                }
             }
-            button7.setOnClickListener {
-                string += "7"
+            buttonMinus.setOnClickListener {
+                if ((string.takeLast(1) !in ".+-") || string == "") {
+                    string += "-"
+                }
                 displayField.text = string
             }
+            buttonPlus.setOnClickListener {
+                if (string.isEmpty()) {
+                    init()
+                } else if (string.takeLast(1) !in "./*+-") {
+                    string += "+"
+                    displayField.text = string
+                }
+            }
+            buttonEquals.setOnClickListener {
+                if (string.isNotEmpty() && string != "-") {
+                    if (string.takeLast(1) in "./*+-") {
+                        string = string.dropLast(1)
+                    }
+                    displayField.text = string
+                    val result = Calculator.result(string)
+                    resultField.text = result
+                    historyList += "$string=$result"
+                } else {
+                    init()
+                }
+            }
+            buttonHist.setOnClickListener {
+                pushFragmentHist(historyList)
+            }
         }
     }
 
     override fun onDestroyView() {
         super.onDestroyView()
         _binding = null
     }
-
-//    open class GetString() {
-//
-//        private var string: String? = null
-//        fun read(): List<String> {
-//
-//        }
-//    }
 }
 
-
-//
-//    println("enter an arithmetic expression without brackets then press 'enter': ")
-//    private var string: String? = readLine()
-//    string = string ?: ""
-//    string = string.replace("""\s+""".toRegex(), "") //all spaces are removed
-//    val operands = mutableListOf<String>()
-//    if (string.isNotBlank()) {
-//        val regex = Regex("""-?\d+(\.\d+)?[*/\-+]?""")
-//        regex.findAll(string).forEach {
-//            operands.add(it.value)
-//        }
-//    }
-//    operands[operands.size - 1] = operands.last().replace("""[*/\-+]?$""".toRegex(), "")
-//    //removing the last symbol of the last operand if action
-//    return operands
-//}
\ No newline at end of file
+class Calculator() {
```

Comment:
можно сделать проще 
```suggestion
object Calculator {
```
тогда все методы (в твоем случае 1) будут статическими + следуя ООП не стоит в одном файле несколько разных сущностей держать